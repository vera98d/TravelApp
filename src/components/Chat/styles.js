import styled from "styled-components";
import { scrollbarStyles } from "../../styles";

export const Container = styled.div`
  background-color: ${(props) => props.theme.colors.secondaryUi};
  height: 600px;
  border-radius: 15px;
  display: flex;
  flex-direction: column;
  box-sizing: border-box;
`;

export const Header = styled.h3`
  text-align: center;
  background-color: ${(props) => props.theme.colors.primaryUi};
  border-radius: 12px 12px 0 0;
  padding: 15px;
  margin: 0;
  font-size: 30px;
  font-weight: 1000;
  font-family: ${(props) => props.theme.fontFamily.secondaryFont};
`;

export const MessagesList = styled.ul`
  display: flex;
  flex-direction: column;
  align-items: flex-start;
  flex-grow: 1;
  margin: 10px;
  padding: 0%;
  overflow: auto;
  ${scrollbarStyles}
`;

export const Footer = styled.div`
  padding: 20px;
  display: flex;
  align-items: center;
`;

export const Input = styled.input`
  flex-grow: 1;
  border: none;
  border-radius: 5px;
  border-style: none;
  margin-right: 10px;
  padding: 10px;
  outline: none;
`;

export const Button = styled.button`
  align-self: center;
  border-radius: 50%;
  border: none;
  height: 30px;
  width: 30px;
  background-color: ${(props) => props.theme.colors.primaryUi};
  display: flex;
  align-items: center;
  justify-content: center;
  cursor: pointer;
  font-family: ${(props) => props.theme.fontFamily.secondaryFont};
`;

export const Message = styled.li`
  list-style: none;
  background-color: ${(props) => props.theme.colors.white};
  border-radius: ${(props) =>
    props.isOwn ? "10px 0 10px 10px" : "0 10px 10px 10px"};
  padding: 15px;
  margin: 5px;
  font-size: 18px;
  font-weight: 500;
  line-height: 22px;
  align-self: ${(props) => (props.isOwn ? "flex-end" : "flex-start")};
`;
