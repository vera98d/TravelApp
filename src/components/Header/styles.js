import styled from "styled-components";

export const Container = styled.div`
  background-color: ${(props) => props.theme.colors.headerBackground};
  display: flex;
  flex-direction: column;
  align-items: center;
  padding-left: 15px;
  padding-right: 15px;
  width: 100%;
`;

export const Title = styled.h1`
  font-family: ${(props) => props.theme.fontFamily.primaryFont};
  font-size: 80px;
  font-weight: 500;
  margin-left: 80px;
  margin-right: 80px;
  padding: 40px 0 0 0;
  text-align: center;
  @media screen and (max-width: 1400px) {
    padding: 20px 0 0 0;
  }
  @media screen and (max-width: 1150px) {
    font-size: 60px;
    padding: 20px 0 0 0;
    line-height: 50px;
  }
  @media screen and (max-width: 560px) {
    width: 350px;
    overflow: hidden;
    text-overflow: ellipsis;
  }
`;

export const Subtitle = styled.h2`
  font-family: ${(props) => props.theme.fontFamily.secondaryFont};
  font-size: 50px;
  font-weight: 500;
  margin: 0;
  padding: 0 0 40px 0;
  text-align: center;
  @media screen and (max-width: 1150px) {
    font-size: 30px;
    padding: 0 0 20px 0;
  }
`;

export const Button = styled.button`
  position: absolute;
  display: flex;
  align-self: flex-start;
  align-items: center;
  justify-content: center;
  border-radius: 50%;
  border: none;
  height: 50px;
  width: 50px;
  background-color: ${(props) => props.theme.colors.tertiaryUi};
  top: 20px;
  left: 10px;
  padding: 0;
  cursor: pointer;
  svg {
    fill: white;
    width: 35px;
    height: 35px;
  }
  @media screen and (max-width: 1150px) {
    height: 40px;
    width: 40px;
    margin: 10px;
    top: 10px;
    svg {
      height: 25px;
      width: 25px;
    }
  }
`;

export const RightButton = styled(Button)`
  left: initial;
  right: 10px;
`;
