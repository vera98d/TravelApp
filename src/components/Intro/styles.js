import styled, { keyframes } from "styled-components";
import img from "../../../src/introImage.jpg";

export const Container = styled.div`
  background-image: url(${img});
  background-size: cover;
  background-repeat: no-repeat;
  background-position: center;
  width: 100vw;
  min-height: 100vh;
  filter: invert(25%);
`;

export const Title1 = styled.h1`
  margin: 0;
  font-family: ${(props) => props.theme.fontFamily.primaryFont};
  font-weight: 400;
  font-size: 300px;
  color: #260101;
  position: absolute;
  left: 70px;
  bottom: 200px;
`;

export const Title2 = styled.h1`
  margin: 0;
  font-family: ${(props) => props.theme.fontFamily.primaryFont};
  font-weight: 400;
  font-size: 220px;
  color: #260101;
  position: absolute;
  left: 80px;
  bottom: 0px;
`;

export const StartButton = styled.div`
  position: fixed;
  display: flex;
  flex-direction: row;
  bottom: 20%;
  right: 20%;
  align-items: center;
  justify-content: center;
  background-color: ${(props) => props.theme.colors.primaryUi};
  color: ${(props) => props.theme.colors.white};
  font-family: ${(props) => props.theme.fontFamily.secondaryFont};
  font-size: 25px;
  padding: 1em 2em;
  border-radius: 10px;
  cursor: pointer;
  transition: 0.3s;

  &:hover {
    background: #df9b89;
    animation-play-state: paused;
  }
`;

const translation = keyframes`
  0%{
      transform: translateX(15px);
  }
  10%{
    transform: translateX(0px);
  }
  20%{
      transform: translateX(15px);
  }
  30%{
    transform: translateX(0px);
  }
  
  100%{
    transform: translateX(0px);
  }
`;

export const Span1 = styled.span`
  color: white;
  display: block;
  margin-left: 1em;
  color: ${(props) => props.theme.colors.white};
  animation: ${translation} 1s 0.3s infinite both ease-in;
  transition: 0.5s;
`;
