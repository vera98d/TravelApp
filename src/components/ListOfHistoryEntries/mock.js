export const entriesMock = [
  {
    id: "1",
    title: "Szlakiem zaginionych",
    city: "Kraków",
  },
  {
    id: "2",
    title: "Trasa Krasnali",
    city: "Wrocław",
  },
  {
    id: "3",
    title: "Piesza wedrówka",
    city: "Góry Sowie",
  },
  {
    id: "4",
    title: "Wycieczka rowerowa",
    city: "Kraków",
  },
  {
    id: "5",
    title: "Narty",
    city: "Zieleniec",
  },
  {
    id: "6",
    title: "Weekend w SPA",
    city: "Wieniec-Zdrój",
  },
  {
    id: "7",
    title: "Może nad morze",
    city: "Mielno",
  },
  {
    id: "8",
    title: "Powrót do rodzinnych stron",
    city: "Przemyśl",
  },
  {
    id: "9",
    title: "Ciemne strony Dolnego Śląska",
    city: "Dolny Śląsk",
  },
  {
    id: "10",
    title: "Zjazd rowerowy ze Ślęzy",
    city: "Sobótka",
  },
  {
    id: "11",
    title: "Rowerem przez wybrzeże",
    city: "Polska",
  },
  {
    id: "12",
    title: "Śladami zwierząt",
    city: "Puszcza Białowieska",
  },
  {
    id: "13",
    title: "W krainie jezior",
    city: "Mazury",
  },
  {
    id: "14",
    title: "Na szczyt",
    city: "Tatry",
  },
];
