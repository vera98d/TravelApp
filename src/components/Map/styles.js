import { MapContainer } from "react-leaflet";
import styled from "styled-components";
import gif from "../../../src/bike.gif";

export const Container = styled.div`
  display: flex;
  justify-content: center;
  padding: 40px 20px;
  background-image: url(${gif});
  background-size: 400px;
  background-repeat: no-repeat;
  background-position: bottom left;
`;

export const StyledMap = styled(MapContainer)`
  width: 600px;
  height: 600px;
  background-color: white;

  a {
    display: none;
  }
`;

export const DefaultProvince = {
  color: `#fff`,
  fillColor: "#BF7D65C4",
  fillOpacity: 1,
  opacity: 1.0,
};

export const HoveredProvince = {
  color: `#fff`,
  fillColor: "#DAE5E5",
  fillOpacity: 0.7,
};
