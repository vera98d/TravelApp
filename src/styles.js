import styled, { createGlobalStyle, css } from "styled-components";

export const GlobalStyle = createGlobalStyle`
  * {
    margin: 0;
    padding: 0;
    box-sizing: border-box;
  }

  @media only screen and (orientation: landscape) and (max-width:750px) {
      padding: 0.5em 1.5em;
    }
  }
`;

export const theme = {
  colors: {
    white: "#fff",
    headerBackground: "#DAE5E5",
    primaryUi: "#BF7D65C4",
    secondaryUi: "#BF7D652B",
    tertiaryUi: "#C4C4C4",
    thirdUi: "rgba(218, 229, 229, 0.26)",
    fourthUi: "#DAE5E5",
  },

  fontFamily: {
    primaryFont: "Bebas Neue, cursive",
    secondaryFont: "Crimson Pro, serif",
  },
};

export const scrollbarStyles = css`
  &::-webkit-scrollbar {
    width: ${(props) => props.width || "10px"};
  }
  &::-webkit-scrollbar-thumb {
    background: ${(props) => props.thumbColor || props.theme.colors.primaryUi};
    border-radius: calc(${(props) => props.width || "10px"} / 2);
  }
  &::-webkit-scrollbar-track {
    background: ${(props) => props.trackColor || props.theme.colors.white};
    border-radius: calc(${(props) => props.width || "10px"} / 2);
  }
`;

export const PrimaryButton = styled.button`
  display: flex;
  align-items: center;
  justify-content: center;
  background-color: ${(props) => props.theme.colors.primaryUi};
  color: ${(props) => props.theme.colors.white};
  font-family: ${(props) => props.theme.fontFamily.secondaryFont};
  font-size: 18px;
  padding: 0.75em 2em;
  border-radius: 10px;
  border: none;
  cursor: pointer;

  &:hover {
    background: #df9b89;
  }
`;
