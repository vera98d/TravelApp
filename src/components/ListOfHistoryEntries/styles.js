import styled from "styled-components";
import { PrimaryButton, scrollbarStyles } from "../../styles";

export const Container = styled.div`
  display: flex;
  flex-direction: column;
  box-sizing: border-box;
  height: 600px;
  max-height: 100%;
  border-radius: 12px;

  div {
    color: grey;
    font-family: ${(props) => props.theme.fontFamily.secondaryFont};
    font-size: 24px;
    padding: 150px 10px;
    height: 100%;
    text-align: center;

    p {
      span {
        text-transform: capitalize;
      }
    }
  }
`;

export const Header = styled.h3`
  font-size: 30px;
  font-weight: 1000;
  font-family: ${(props) => props.theme.fontFamily.secondaryFont};
  text-align: center;
  background-color: ${(props) => props.theme.colors.primaryUi};
  border-radius: 12px;
  padding: 15px;
  margin-bottom: 20px;
`;

export const EntriesList = styled.ul.attrs((props) => ({
  width: "20px",
  trackColor: props.theme.colors.thirdUi,
  thumbColor: props.theme.colors.fourthUi,
}))`
  display: flex;
  flex-direction: column;
  align-items: flex-start;
  flex-grow: 1;
  padding: 0%;
  overflow: auto;
  ${scrollbarStyles}
`;

export const Entry = styled.li`
  list-style: none;
  background-color: ${(props) => props.theme.colors.secondaryUi};
  font-family: ${(props) => props.theme.fontFamily.secondaryFont};
  font-size: 18px;
  font-style: italic;
  width: 90%;
  padding: 10px;
  margin: 5px 0;
  border: 2px solid transparent;
  border-radius: 10px;
  cursor: pointer;

  &:hover {
    border: 2px solid ${(props) => props.theme.colors.primaryUi};
    box-shadow: 0 2px 5px gray;
  }
`;
export const Button = styled(PrimaryButton)`
  align-self: flex-start;
  margin-top: 30px;
`;
