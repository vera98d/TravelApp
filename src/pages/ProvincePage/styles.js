import styled from "styled-components";
import { Chat } from "../../components/Chat/Chat";
import { ListOfHistoryEntries } from "../../components/ListOfHistoryEntries";

export const Container = styled.div`
  display: flex;
  align-items: center;
  flex-direction: column;
`;

export const Wrapper = styled.div`
  display: flex;
  justify-content: space-around;
  margin: 0 auto;
  padding: 40px 20px;
  max-width: 1500px;
  width: 100%;
  gap: 60px;
  @media screen and (max-width: 1150px) {
    gap: 30px;
  }
  @media screen and (max-width: 640px) {
    flex-direction column;
  }
`;

export const StyledListOfHistoryEntries = styled(ListOfHistoryEntries)`
  width: 35%;
  @media screen and (max-width: 640px) {
    width: 100%;
  }
`;

export const StyledChat = styled(Chat)`
  width: 65%;
  @media screen and (max-width: 640px) {
    width: 100%;
  }
`;
