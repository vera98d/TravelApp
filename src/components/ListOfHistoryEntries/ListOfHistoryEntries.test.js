import { render, screen } from "@testing-library/react";
import { ThemeProvider } from "styled-components";
import { theme } from "../../styles";
import { ListOfHistoryEntries } from "./index";
import historyEntriesService from "../../services/HistoryEntriesService";
import { entriesMock } from "./mock";

let orginalGetAll;
beforeEach(() => {
  orginalGetAll = historyEntriesService.getAll;
  historyEntriesService.getAll = () => Promise.resolve(entriesMock);
});

afterEach(() => {
  historyEntriesService.getAll = orginalGetAll;
});

const entryTitle = {
  title: "Szlakiem krasnali",
  city: "Wroclaw",
};

describe("ListOfHistoryEntries", () => {
  test("matches snapshot", async () => {
    const { container } = render(
      <ThemeProvider theme={theme}>
        <ListOfHistoryEntries props={entryTitle} />
      </ThemeProvider>
    );
    await screen.findAllByRole("listitem");
    expect(container).toMatchSnapshot();
  });

  test('A "Create new story" button is rendered', () => {
    render(
      <ThemeProvider theme={theme}>
        <ListOfHistoryEntries props={entryTitle} />
      </ThemeProvider>
    );

    const button = screen.getByText("Create new story");
    expect(button).toBeInTheDocument();
  });

  test("extracts the button DOM node", () => {
    render(
      <ThemeProvider theme={theme}>
        <ListOfHistoryEntries props={entryTitle} />
      </ThemeProvider>
    );

    const button = screen.getByRole("button");
    expect(button).not.toBeDisabled();
  });

  test('Header should show "History of entries"', () => {
    render(
      <ThemeProvider theme={theme}>
        <ListOfHistoryEntries props={entryTitle} />
      </ThemeProvider>
    );

    const header = screen.queryByText("History of entries");
    expect(header).toBeInTheDocument();
  });

  test("is rendering all entries in each province", async () => {
    render(
      <ThemeProvider theme={theme}>
        <ListOfHistoryEntries props={entryTitle} />
      </ThemeProvider>
    );

    const lists = await screen.findAllByRole("listitem");
    expect(lists.length).toBe(entriesMock.length);
  });
});
