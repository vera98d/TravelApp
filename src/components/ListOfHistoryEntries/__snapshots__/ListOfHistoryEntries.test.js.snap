// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ListOfHistoryEntries matches snapshot 1`] = `
<div>
  <div
    class="sc-gsDKAQ gTnIr"
  >
    <h3
      class="sc-dkPtRN fkbcZJ"
    >
      History of entries
    </h3>
    <ul
      class="sc-hKwDye eMHPPf"
      width="20px"
    >
      <li
        class="sc-eCImPb jUhYUa"
      >
        Szlakiem zaginionych
         – 
        Kraków
      </li>
      <li
        class="sc-eCImPb jUhYUa"
      >
        Trasa Krasnali
         – 
        Wrocław
      </li>
      <li
        class="sc-eCImPb jUhYUa"
      >
        Piesza wedrówka
         – 
        Góry Sowie
      </li>
      <li
        class="sc-eCImPb jUhYUa"
      >
        Wycieczka rowerowa
         – 
        Kraków
      </li>
      <li
        class="sc-eCImPb jUhYUa"
      >
        Narty
         – 
        Zieleniec
      </li>
      <li
        class="sc-eCImPb jUhYUa"
      >
        Weekend w SPA
         – 
        Wieniec-Zdrój
      </li>
      <li
        class="sc-eCImPb jUhYUa"
      >
        Może nad morze
         – 
        Mielno
      </li>
      <li
        class="sc-eCImPb jUhYUa"
      >
        Powrót do rodzinnych stron
         – 
        Przemyśl
      </li>
      <li
        class="sc-eCImPb jUhYUa"
      >
        Ciemne strony Dolnego Śląska
         – 
        Dolny Śląsk
      </li>
      <li
        class="sc-eCImPb jUhYUa"
      >
        Zjazd rowerowy ze Ślęzy
         – 
        Sobótka
      </li>
      <li
        class="sc-eCImPb jUhYUa"
      >
        Rowerem przez wybrzeże
         – 
        Polska
      </li>
      <li
        class="sc-eCImPb jUhYUa"
      >
        Śladami zwierząt
         – 
        Puszcza Białowieska
      </li>
      <li
        class="sc-eCImPb jUhYUa"
      >
        W krainie jezior
         – 
        Mazury
      </li>
      <li
        class="sc-eCImPb jUhYUa"
      >
        Na szczyt
         – 
        Tatry
      </li>
    </ul>
    <button
      class="sc-bdvvtL sc-jRQBWg eHPgEv icFjEr"
      type="button"
    >
      Create new story
    </button>
  </div>
</div>
`;
